ocaml/typing/typecore.ml

```diff
@@ -5928,8 +5928,10 @@ and type_expect_
         exp_type = ty';
         exp_attributes = arg.exp_attributes;
         exp_env = env;
-        exp_extra = (Texp_constraint (Some extra_cty, modes), loc, sexp.pexp_attributes)
-                    :: arg.exp_extra;
+        exp_extra =
+          (Texp_constraint (Some extra_cty, Typemode.transl_mode_annots modes),
```

Comment:
`transl_mode_annots` is called twice - we should factor that out. Not just for performance reasons, but also it's more "correct"